### PR TITLE
feat: add neonbee-kickstart-guide

### PR DESCRIPTION
**Pull request by pk-work**

### CODE REVIEW

File: neonbee-kickstart-guide/README.md [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+import io.vertx.junit5.VertxTestContext;
+
+class PopulationDataVerticleTest extends DataVerticleTestBase {
+
+    @BeforeEach
+    public void setUp(VertxTestContext testContext) {
+        // The PopulationDataVerticle has a dependency on the BeehivesDataVerticle. In order to have full control
+        // over the return value of the BeehivesDataVerticle we need to mock it.
+        DataVerticle<JsonArray> dummyBeehives =
+                createDummyDataVerticle(QUALIFIED_NAME).withStaticResponse(new JsonArray().add(BEEHIVE_1));
```
Comment (s4heid) [inline neonbee-kickstart-guide/README.md:332]: nit; no static import to make it more explicit which verticle's qualified name you are using

File: neonbee-kickstart-guide/README.md [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+   entity Beehives {
+      key id : Integer;
+      queen : String;
+      drones : Integer;
+      worker : Integer;
+      population : Integer;
+   }
+}
+```
+
```
Comment (s4heid) [inline neonbee-kickstart-guide/README.md:397]: maybe add some explanation of the example model, e.g.:

```suggestion
A CDS model is a way of organizing and defining the data and relationships within a service. It specifies the entities, their fields, and any relationships between them. In this case, the model consists of the `BeehiveService` and the `Beehives` entity, and defines the structure and content of the data stored in the service. The Beehives entity has a key field called "id", which is an integer, and four other fields: "queen", "drones", "worker", and "population". These fields represent different aspects of a beehive, such as the type of bees present or the size of the hive's population.
```

File: neonbee-kickstart-guide/README.md [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+## Part 3: The EntityVerticle
+
+In this section you will write your first *EntityVerticle* and expose its data through the structured OData endpoint.
+But before you can start, you need to ...
+
+1. ... add a models subproject to your gradle project.
+2. ... define the OData model of your data in CDS.
```
Comment (s4heid) [inline neonbee-kickstart-guide/README.md:369]:
```suggestion
2. ... define the OData model of your data in [Common Data Service (CDS)](https://cap.cloud.sap/docs/cds/).
```

File: neonbee-kickstart-guide/README.md [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+      drones : Integer;
+      worker : Integer;
+      population : Integer;
+   }
+}
+```
+
+### Write your first EntityVerticle
+
+The purpose of *EntityVerticles* is to process structured data. Now that you have defied a data structure in the CDS file,
```
Comment (s4heid) [inline neonbee-kickstart-guide/README.md:400]: typo
```suggestion
The purpose of *EntityVerticles* is to process structured data. Now that you have defined a data structure in the CDS file,
```

File: neonbee-kickstart-guide/gradle/checkstyle/test.xml [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            <property name="allowMultipleEmptyLinesInsideClassMembers" value="false" />
+            <property name="allowNoEmptyLineBetweenFields" value="true" />
+        </module>
+        <module name="GenericWhitespace" />
+        <module name="MethodParamPad" />
+        <module name="NoWhitespaceBefore" />
+        <module name="ParenPad" />
+        <module name="OperatorWrap" />
+
+        <!-- JavaDoc Checks See http://checkstyle.org/config_javadoc.html -->
```
Comment (s4heid) [inline neonbee-kickstart-guide/gradle/checkstyle/test.xml:144]: i guess you can delete this?

File: neonbee-kickstart-guide/gradlew [hunk 6]
```diff
@@ -0,0 +1,234 @@
+#!/bin/sh
+
+#
+# Copyright © 2015-2021 the original authors.
```
Comment (s4heid) [inline neonbee-kickstart-guide/gradlew:4]: update license year?
```suggestion
# Copyright © 2015-2023 the original authors.
```

**Review by s4heid — CHANGES_REQUESTED** [review]
cool code examples! just some minor things I noticed.

File: neonbee-kickstart-guide/gradle/checkstyle/test.xml (hunk 5 above)
Comment (pk-work) [reply neonbee-kickstart-guide/gradle/checkstyle/test.xml:144]: Yes, but it was added by the plugin.

File: neonbee-kickstart-guide/gradlew (hunk 6 above)
Comment (pk-work) [reply neonbee-kickstart-guide/gradlew:4]: This file was generated by gradle, so I will not touch it.

**Pull request merged**